packages/tools/examples/freehandROISculptorTool/index.ts

```diff
@@ -0,0 +1,283 @@
+} = cornerstoneTools;
+
+const { ViewportType } = Enums;
+const { MouseBindings, Events } = csToolsEnums;
+
+// Define a unique id for the volume
+const volumeName = 'CT_VOLUME_ID'; // Id of the volume less loader prefix
+const volumeLoaderScheme = 'cornerstoneStreamingImageVolume'; // Loader id which defines which volume loader to use
+const volumeId = `${volumeLoaderScheme}:${volumeName}`; // VolumeId with loader id + volume id
+const renderingEngineId = 'myRenderingEngine';
+const viewportIds = ['CT_STACK', 'CT_VOLUME_SAGITTAL'];
+
+// ======== Set up page ======== //
+setTitleAndDescription(
+  'Freehand ROI Sculptor Tool',
+  'Here we demonstrate how to use the Freehand ROI Sculptor Tool to sculpt freehand ROIs'
+);
+
+const size = '500px';
+const content = document.getElementById('content');
+const viewportGrid = document.createElement('div');
+
+viewportGrid.style.display = 'flex';
+viewportGrid.style.display = 'flex';
+viewportGrid.style.flexDirection = 'row';
+
+const element1 = document.createElement('div');
+const element2 = document.createElement('div');
+element1.style.width = size;
+element1.style.height = size;
+element2.style.width = size;
+element2.style.height = size;
+
+// Disable right click context menu so we can have right click tool
+element1.oncontextmenu = (e) => e.preventDefault();
+element2.oncontextmenu = (e) => e.preventDefault();
+
+viewportGrid.appendChild(element1);
+viewportGrid.appendChild(element2);
+
+content.appendChild(viewportGrid);
+
+const instructions = document.createElement('p');
+instructions.innerText = `
+Drawing:
+
+- Click on FreeHandROI button and drag to draw a freehand annotation.
+- If you join the contour together it will be closed, otherwise releasing the mouse will create an open contour (freehand line).
+- Click on LivewireContour button to draw LivewireContour annotations
+
+Editing:
+
+- Click on Sculpt button then adjustable cursor will appear.
+- Nearest freehand ROI will be selected while clicking, and toolsize can be adjusted by moving cursor near to selected annotation
+- Drag to sculpt freehand ROI
+
+`;
+
+content.append(instructions);
+
+function addFreeHandRoiToolButton(toolGroup) {
+  addButtonToToolbar({
+    title: 'FreeHandROI',
+    onClick: () => {
+      toolGroup.setToolActive(PlanarFreehandROITool.toolName, {
+        bindings: [
+          {
+            mouseButton: MouseBindings.Primary, // Left Click
+          },
+        ],
+      });
+      toolGroup.setToolPassive(FreehandROISculptorTool.toolName);
+      toolGroup.setToolPassive(LivewireContourTool.toolName);
+    },
+  });
+}
+
+function addSculptingToolButton(toolGroup) {
+  addButtonToToolbar({
+    title: 'Sculpt',
+    onClick: () => {
+      toolGroup.setToolActive(FreehandROISculptorTool.toolName, {
+        bindings: [
+          {
+            mouseButton: MouseBindings.Primary, // Left Click
+          },
+        ],
+      });
+      toolGroup.setToolPassive(PlanarFreehandROITool.toolName);
```

Comment:
This is a bit disturbing here to have to set these to passive.  The normal set tool active should deal with setting other tools using the same binding to be setup appropriately.  At very least you need documentation on why this is getting set the way it is, and ideally some configuration which could be set on the tool to allow automating this without necessarily needing to know what all tool groups need to be updated.

Reply:
You are just setting all the other tools passive here, rather than knowing which one was previously active.  Could
you follow the standard pattern and keep track of which tool is active and update that setting as required?  That way
you will be able to support more tools than just these ones, or more configurations for them - like spline.

Reply:
Updated